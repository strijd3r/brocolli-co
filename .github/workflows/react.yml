# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: Node.js CI

on:
  push:
  pull_request:

jobs:
  build:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14.x]

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2

      - name: Set up Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}

      - name: Install dependencies
        run: yarn

      - name: Run the tests
        run: yarn test

      - name: Build
        run: yarn build
        if: startsWith(github.ref, 'refs/tags/')

      - name: Release
        uses: softprops/action-gh-release@v1
        if: startsWith(github.ref, 'refs/tags/')
        with:
          files: |
            README.md
            package.json
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

      - name: Deploy
        uses: peaceiris/actions-gh-pages@v3
        if: startsWith(github.ref, 'refs/tags/')
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./build
