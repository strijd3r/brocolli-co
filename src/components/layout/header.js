import React, { Component } from 'react';
import styled from 'styled-components';
import { darken } from 'polished';
import logo from './header.logo.svg';

/**
 * HeaderComponent
 * @param {object} props HeaderComponent properties
 * @param {string} props.children Children
 * @param {string} props.className Classname
 * @returns
 */
function HeaderComponent({ children, className }) {
  return (
    <header className={className}>
      <Logo src={logo} />
      {children}
    </header>
  );
}

/**
 * Logo is a styled image with a default height
 */
export const Logo = styled.img`
  height: 4rem;
`;

Logo.defaultProps = {
  src: logo,
};

/**
 * Header is a styled version of the HeaderComponent.
 * @type {Component}
 */
export const Header = styled(HeaderComponent)`
  height: 4rem;
  background: ${(props) => props.theme.primaryBg};
  color: ${(props) => props.theme.primary};
  border: 2px solid ${(props) => darken(0.1, props.theme.primaryBg)};
  box-shadow: 5px 5px 0 ${(props) => darken(0.1, props.theme.primaryBg)};
`;

Header.defaultProps = {
  theme: {
    primary: '#000000',
    primaryBg: '#FFFFFF',
  },
};
