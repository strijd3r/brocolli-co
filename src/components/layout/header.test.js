import { render, screen } from '@testing-library/react';
import 'jest-styled-components'

import { Header } from './header';

test('renders div and its children', () => {
  const header = "testing react header"
  const { container } = render(<Header>{header}</Header>);
  expect(container.firstChild).toMatchSnapshot()

  const headerEl = screen.getByText(header);
  expect(headerEl).toBeInTheDocument();
});
