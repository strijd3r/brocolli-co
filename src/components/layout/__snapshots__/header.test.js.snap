// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders div and its children 1`] = `
<header
  class=""
>
  <img
    src="header.logo.svg"
  />
  testing react header
</header>
`;
